File: regression/python/class9/main.c
#include <string.h>
#include <assert.h>

struct BankAccount
{
  const char *owner_;
};

#if 0
struct BankAccount init(const char *owner)
{
  struct BankAccount ba;
  ba.owner_ = owner;
  return ba;
}

int main(void)
{
  struct BankAccount b = init("Alice");
  assert(strcmp(b.owner_, "Alice") == 0);
  return 0;
}
#endif

void foo(const char *ptr)
{
  const char *copy = ptr;
//  assert(strcmp("bruno", copy) == 0);
}

int main(void)
{
  struct BankAccount ba = {.owner_ = "bruno"};

//code
//  * type: code
//  * operands:
//    0: symbol
//        * type: symbol
//            * identifier: tag-struct BankAccount
//        * name: ba
//        * identifier: c:main.c@438@F@main@ba
//    1: struct
//        * type: struct
//            * tag: struct BankAccount
//            * components:
//              0: component
//                  * type: pointer
//                      * subtype: signedbv
//                          * width: 8
//                          * #constant: 1
//                          * #cpp_type: signed_char
//                  * name: owner_
//                  * pretty_name: owner_
//                  * #location:
//                    * file: main.c
//                    * line: 6
//                    * column: 3
//            * #location:
//              * file: main.c
//              * line: 4
//              * column: 1
//        * operands:
//          0: address_of
//              * type: pointer
//                  * subtype: signedbv
//                      * width: 8
//                      * #cpp_type: signed_char
//              * operands:
//                0: index
//                    * type: signedbv
//                        * width: 8
//                        * #cpp_type: signed_char
//                    * operands:
//                      0: string-constant
//                          * type: array
//                              * size: constant
//                                  * type: unsignedbv
//                                      * width: 64
//                                  * value: 0000000000000000000000000000000000000000000000000000000000000110
//                                  * #cformat: 6
//                              * subtype: signedbv
//                                  * width: 8
//                                  * #cpp_type: signed_char
//                          * value: bruno
//                          * kind: default
//                          * #location:
//                            * file: main.c
//                            * line: 33
//                            * function: main
//                            * column: 38
//                      1: constant
//                          * type: signedbv
//                              * width: 64
//                          * value: 0000000000000000000000000000000000000000000000000000000000000000
//              * #location:
//                * file: main.c
//                * line: 33
//                * function: main
//                * column: 38
//        * #location:
//          * file: main.c
//          * line: 33
//          * function: main
//          * column: 27
//  * statement: decl
//  * #location:
//    * file: main.c
//    * line: 33
//    * function: main
//    * column: 3

  foo(ba.owner_);

//  member
//    * type: pointer
//        * subtype: signedbv
//            * width: 8
//            * #constant: 1
//            * #cpp_type: signed_char
//    * operands:
//      0: symbol
//          * type: symbol
//              * identifier: tag-struct BankAccount
//          * name: ba
//          * identifier: c:main.c@438@F@main@ba
//          * #location:
//            * file: main.c
//            * line: 34
//            * function: main
//            * column: 7
//          * #lvalue: 1
//    * component_name: owner_
//    * #location:
//      * file: main.c
//      * line: 34
//      * function: main
//      * column: 7

//  assert(strcmp("bruno", ba.owner_) == 0);
  return 0;
}

File: regression/python/class9/main.py
class BankAccount:
    def __init__(self, owner: str):
        self.owner = owner
        
account = BankAccount("Alice")
assert account.owner == "Alice"

#strcmp call:

# sideeffect
#   * type: signedbv
#       * width: 32
#   * operands: 
#     0: symbol
#         * type: code
#             * arguments: 
#               0: argument
#                   * type: pointer
#                       * subtype: signedbv
#                           * width: 8
#                           * #constant: 1
#                           * #cpp_type: signed_char
#                   * #location: 
#                     * file: /home/bruno/projects/esbmc/src/c2goto/library/string.c
#                     * line: 94
#                     * function: strcmp
#                     * column: 12
#                   * #base_name: p1
#                   * #identifier: c:string.c@1408@F@strcmp@p1
#               1: argument
#                   * type: pointer
#                       * subtype: signedbv
#                           * width: 8
#                           * #constant: 1
#                           * #cpp_type: signed_char
#                   * #location: 
#                     * file: /home/bruno/projects/esbmc/src/c2goto/library/string.c
#                     * line: 94
#                     * function: strcmp
#                     * column: 28
#                   * #base_name: p2
#                   * #identifier: c:string.c@1424@F@strcmp@p2
#             * return_type: signedbv
#                 * width: 32
#                 * #cpp_type: signed_int
#         * name: strcmp
#         * identifier: c:@F@strcmp
#     1: arguments
#         * operands: 
#           0: member
#               * type: pointer
#                   * subtype: signedbv
#                       * width: 8
#               * operands: 
#                 0: symbol
#                     * type: symbol
#                         * identifier: tag-BankAccount
#                     * identifier: py:main.py@account
#               * component_name: owner
#           1: address_of
#               * type: pointer
#                   * subtype: signedbv
#                       * width: 8
#               * operands: 
#                 0: index
#                     * type: signedbv
#                         * width: 8
#                     * operands: 
#                       0: constant
#                           * type: array
#                               * size: constant
#                                   * type: unsignedbv
#                                       * width: 64
#                                   * value: 0000000000000000000000000000000000000000000000000000000000000101
#                                   * #cformat: 5
#                               * subtype: signedbv
#                                   * width: 8
#                           * operands: 
#                             0: constant
#                                 * type: signedbv
#                                     * width: 8
#                                 * value: 01000001
#                                 * #cformat: 65
#                             1: constant
#                                 * type: signedbv
#                                     * width: 8
#                                 * value: 01101100
#                                 * #cformat: 108
#                             2: constant
#                                 * type: signedbv
#                                     * width: 8
#                                 * value: 01101001
#                                 * #cformat: 105
#                             3: constant
#                                 * type: signedbv
#                                     * width: 8
#                                 * value: 01100011
#                                 * #cformat: 99
#                             4: constant
#                                 * type: signedbv
#                                     * width: 8
#                                 * value: 01100101
#                                 * #cformat: 101
#                       1: constant
#                           * type: signedbv
#                               * width: 64
#                           * value: 0000000000000000000000000000000000000000000000000000000000000000
#   * statement: function_call
